packages/plugin-node/src/rules/consoleSpaces.ts

```diff
@@ -0,0 +1,74 @@
+import { getTSNodeRange, typescriptLanguage } from "@flint.fyi/ts";
+import * as ts from "typescript";
+
+const CONSOLE_METHODS = new Set([
+	"assert",
+	"count",
+	"countReset",
+	"debug",
+	"dir",
+	"dirxml",
+	"error",
+	"group",
+	"groupCollapsed",
+	"info",
+	"log",
+	"table",
+	"time",
+	"timeEnd",
+	"timeLog",
+	"trace",
+	"warn",
+]);
+
+function hasLeadingOrTrailingSpaces(text: string): boolean {
+	return text.length > 0 && (text.startsWith(" ") || text.endsWith(" "));
+}
+
+export default typescriptLanguage.createRule({
+	about: {
+		description:
+			"Disallow leading or trailing spaces in console method string arguments.",
+		id: "consoleSpaces",
+		preset: "stylistic",
+	},
+	messages: {
+		noConsoleSpaces: {
+			primary: "Avoid leading or trailing spaces in console method arguments.",
+			secondary: [
+				"Leading or trailing spaces in console output are often unintentional and can make debugging harder.",
+				"Use explicit spacing in the format string or separate arguments instead.",
+			],
+			suggestions: ["Remove leading and trailing spaces from string literals"],
+		},
```

Comment:
Split this into two reports:

* Leading spaces
* Trailing spaces

...and update the implementation & tests to reflect this.